### DISCUSSION TITLE
Add Svelte support

### DISCUSSION THREAD

**Discussion by avivash** [rainbow-me/rainbowkit#395] [Ideas]
With the [growing popularity of Svelte over the past couple years](https://javascript.plainenglish.io/the-most-popular-javascript-frameworks-in-2021-a2fe62174df6), it would be cool to explore building a version of RainbowKit that supports Svelte. I'm sure the community(myself included) would be willing to help with this 👍🏼

**Comment by nickbytes** [comment]
Thanks for the feedback! I like this idea. We would definitely be supportive of any Svelte (or Vue) versions of RainbowKit. 

We currently do not have the time to invest in this ourselves, but if you and the community want to create an RFC/proposal and begin work here, we're happy to review and provide some guidance. This could even start as a fork, and then we could figure out if it makes sense/is feasible to add to our monorepo.

**Reply by avivash** [reply]
Awesome! I've actually already started building it out in my free time(and when I'm not doing things specifically related to my day job), but I'm happy to write up an RFC for it 👍🏼 I'll make sure to fork it too, so it'll be easier to merge back in down the road if we decide that's the way to go. 

It looks like [@wagmi/core](https://github.com/tmm/wagmi/tree/main/packages/core) exposes everything we need to get the functionality of `@wagmi` in a non-React-specific way too, so I don't think that'll be a blocker. I'm excited to dig deeper into this 🙌🏼

**Reply by hardyjosh** [reply]
@avivash another Svelte user over here - let me know if you want to collaborate on this!

**Reply by multiplehats** [reply]
Down to help a hand too @avivash .

**Reply by Dhaiwat10** [reply]
Down to help on this too @avivash!

**Reply by tony-stark-eth** [reply]
Down to help too! @avivash

**Reply by NikAgs** [reply]
@avivash How's development coming? Is there a repo we can make contributions to?

**Reply by softwarecurator** [reply]
looking to help too

**Reply by LyghtCode** [reply]
same

**Reply by dionysuzx** [reply]
same

**Reply by LyghtCode** [reply]
> Awesome! I've actually already started building it out in my free time(and when I'm not doing things specifically related to my day job), but I'm happy to write up an RFC for it 👍🏼 I'll make sure to fork it too, so it'll be easier to merge back in down the road if we decide that's the way to go.
> 
> It looks like [@wagmi/core](https://github.com/tmm/wagmi/tree/main/packages/core) exposes everything we need to get the functionality of `@wagmi` in a non-React-specific way too, so I don't think that'll be a blocker. I'm excited to dig deeper into this 🙌🏼

is there a repo in progress we can help with?

**Reply by parseb** [reply]
wen? any updates?

**Reply by gh-osama** [reply]
any repo here we can help with?

**Reply by Olyno** [reply]
Given the situation, either someone takes the lead and creates the repo, or we will never have this integration.

Also, I would like to suggest making the project as a web component, in order to make it compatible with all existing frameworks and not only Svelte.

**Reply by sameoldlab** [reply]
I've got time to try working on this the next few months. But just to clarify, do people primarily want:
- 👀:  a UI component with accessibility, responsiveness, etc already done
- 🚀: or javascript/svelte behaviours to manage connect, signing, state

js behavours don't seem like they should be too hard with svelte stores. I could start with the UI in web components, will be figuring out good architecture along the way. There are also plugins to create web components from react.

**Reply by hardyjosh** [reply]
I think a big part of what Rainbowkit is offering is a UI that looks good out of the box.

For ethers compatible stores, I've used https://gitlab.com/clb1/svelte-ethers-store on every project, it has a nice API. Not sure if it's useful for this directly but could be a source of inspiration.

**Reply by softwarecurator** [reply]
using `wagmi/core` seems like a better alternative then using that and building a new store around wagmi to keep the rainbowkit flow

**Reply by sameoldlab** [reply]
@hardyjosh that’s what I thought. Assuming web components don’t have some dirty implementation bug, that could cover UI side for svelte and vue requests. And possibly part of the solana and polkadot ones as well. Then let everyone write whatever idiomatic glue is needed for their framework/blockchain.
Maybe this should be a new discussion post?

**Reply by etashhh** [reply]
@sameoldlab an out-of-the-box UI component would indeed be very helpful 🤝

**Reply by dionysuzx** [reply]
I built a dapp starter with svelte and web3modal here: https://github.com/d1onys1us/dapp-slaps

Hope it helps anyone wanting to build with svelte+foundry stack in a monorepo. let's not give up on using svelte for web3, react is just not pragmatic (source: https://component-party.dev/). we can make svelte popular in this space 🤝

**Reply by sameoldlab** [reply]
I made some progress here https://github.com/sameoldlab/fractl-ui. Built it with svelte, but using the web component export. It's not directly a port of rainbowkit, because I'm also figuring out a multi-chain setup and would want to talk to someone at rainbow first. 
But if anyone has time to help out or kick the tires pls lmk.
cc: @etashhh @hardyjosh

**Comment by laciferin2024** [comment]
Svelte 5 is stable now, please make considerable effort to support
